baseapp/abci.go

```diff
@@ -668,16 +634,23 @@ func (app *BaseApp) FinalizeBlock(_ context.Context, req *abci.RequestFinalizeBl
 		app.finalizeBlockState.ms = app.finalizeBlockState.ms.SetTracingContext(nil).(storetypes.CacheMultiStore)
 	}
 
-	endBlockResp := app.legacyEndBlock(req)
+	valsetupdate, endEvents, err := app.Endblock(app.finalizeBlockState.ctx)
```

Comment:
I don't understand this diff. There is no reason to move what I had into a (private) `legacyEndBlocker` to a public `EndBlock`. Just modify the existing `legacyEndBlock`. I'm OK with removing `legacy` from the method name, but there is no reason for it to be public.